src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -132,17 +131,120 @@ def __init__(self, *args, **kwargs):
         # use 'add_cancel_ticket(<ticket>)' to add to this set.
         self._teardown_cancel = set()
         self._teardown_errors = []
+        self.basepath = None
+        self.prefix = None
+        self.ofi_prefix = None
 
     def setUp(self):
         """Set up each test case."""
+        # get paths from the build_vars generated by build
+        with open('../../.build_vars.json') as build_vars:
+            build_paths = json.load(build_vars)
+        self.basepath = os.path.normpath(os.path.join(build_paths['PREFIX'],
+                                                      '..') + os.path.sep)
+        self.prefix = build_paths['PREFIX']
+        try:
+            self.ofi_prefix = build_paths['OFI_PREFIX']
+        except KeyError:
+            self.ofi_prefix = os.sep + "usr"
+        self.cancel_from_list()
         self.check_variant_skip()
         self.log.info("*** SETUP running on %s ***", str(detect()))
         super(Test, self).setUp()
 
-    def check_variant_skip(self):
+    def add_test_data(self, filename, data):
+        """Add a file containing the specified data to the test variant
+           specific data directory.
+
+        Args:
+            filename (str): name of the file to create
+            data (list): data to write to the new file
+        """
+        try:
+            with open(os.path.join(self.outputdir, filename),
+                      "w") as file_handle:
+                file_handle.writelines(data)
+        except IOError as error:
+            self.fail("Error writing {}: {}".format(filename, error))
+
+    def cancel_from_list(self):
+        """Check if test is in skip list"""
+
+        def skip_process_error(message):
+            self.log.info(message)
+            self.log.info("Trudging on without skipping known failing tests")
+
+        def cancelForTicket(ticket, skip_list):
```

Comment:
(pylint-invalid-name) Function name "cancelForTicket" doesn't conform to snake_case naming style